Sources/Kitura/RouterParameterWalker.swift

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright IBM Corporation 2016
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+public typealias RouterParameterHandler = (RouterRequest, RouterResponse, String, @escaping () -> Void) throws -> Void
+
+class RouterParameterWalker {
+
+    ///
+    private var parameterHandlers: [String : [RouterParameterHandler]]
+
+    init(handlers: [String : [RouterParameterHandler]]) {
+        self.parameterHandlers = handlers
+    }
+
+    func handle(request: RouterRequest, response: RouterResponse, with callback: @escaping () -> Void) {
+        let filtered = request.parameters.filter { (key, _) in self.parameterHandlers.keys.contains(key) }
+        self.handle(filtered: filtered, request: request, response: response, with: callback)
+    }
+
```

Comment:
It took a while to understand what was going on here.

If I understand correctly, you are modifying the copied arrays of parameters and parameterHandlers in order to do the recursion and/or looping without an index (i.e. other state in the RouterParameterWalkser class.

I have two concerns here:
    1. Performance. Which performs better the algorithm in the RouterElementWalker class, which uses an index, albeit captures a local variable or this one which is mdifying, thus copying the arrays.
    2. Consistency. I would have liked all of the "walker" classes to be similar in nature. It makes it easier to maintain them.